test/test_jit.py

```diff
@@ -5041,6 +5041,34 @@ def test(op, args):
         for op, lhs, rhs in product(ops, type_literals, type_literals):
             test(op, [lhs, rhs])
 
+    def test_isinstance(self):
```

Comment:
Could you add a test for isinstance(x: Optional[int], int) ? 

I don't know that we need to support that behavior, but maybe throw an error here ?

Reply:
added the test case with error throwing for optional types